lib/models/repository/genre_repository.dart

```diff
@@ -0,0 +1,36 @@
+import 'dart:async';
+import 'dart:convert';
+import 'package:flutter/services.dart';
+import 'package:the_movie_db/utils/general_constants.dart';
+import '../genre.dart';
+import 'repository.dart';
+
+class GenreRepository implements Repository {
+  static const String jsonFilePath = 'data/genres.json';
+  static List<Genre> genres = [];
+
+  @override
+  Future<void> fetchData() async {
```

Comment:
I wouldn't add this fetchData method as an interface method. You only need a get method, the implementation will take care of the where you get the data.
Same for the movies repo

Reply:
done